src/components/WeeklySummariesReport/WeeklySummariesReport.jsx

```diff
@@ -22,6 +24,20 @@ export class WeeklySummariesReport extends Component {
 
   async componentDidMount() {
```

Comment:
the asynchronous code in the componentDidMount function could potentially lead to performance issues. It sends a request for each summary, and if there are a large number of summaries, this could result in a large number of requests, which could put significant pressure on the server.